testing/unittests/unittests.cpp

```diff
@@ -569,5 +570,58 @@ class StringBufferTest : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( StringBufferTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( StringBufferTest, "StringBufferTest" );
 
+StringBuffer &mbToBase58(StringBuffer &s, const MemoryBuffer &data)
+{
+    size_t b58Length = data.length() * 2;
+
+    if (b58enc(s.reserve(b58Length), &b58Length, data.toByteArray(), data.length()) && b58Length > 1)
+        s.setLength(b58Length);
+    return s;
+}
+
+StringBuffer &base64ToBase58(StringBuffer &s, const char *b64)
+{
+    MemoryBuffer mb;
+    JBASE64_Decode(b64, mb);
+    return mbToBase58(s, mb);
+}
+
+StringBuffer &textToBase58(StringBuffer &s, const char *text)
+{
+    MemoryBuffer mb;
+    mb.append((size_t)strlen(text), text);
+    return mbToBase58(s, mb);
+}
+
+class Base58Test : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE( Base58Test  );
+        CPPUNIT_TEST(testEncode);
+    CPPUNIT_TEST_SUITE_END();
+
+    void testEncode()
+    {
+        StringBuffer s;
+
+        //short string
+        ASSERT(streq(textToBase58(s.clear(), "1"), "r"));
```

Comment:
Good test.
Worth testing large character values as well.  e.g. "\xff", which unfortunately fails.

Reply:
Also good to test on a zero length string - which returns false

Reply:
I'll add a couple more tests, and see about the \xff failure and weather it matters.